app/gui/src/dashboard/utilities/tanstackQuery.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file A collection of utility functions for Tanstack Query.
+ */
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
+import {
+  useQueryClient,
+  type DefaultError,
+  type EnsureQueryDataOptions,
+  type QueryClient,
+  type QueryKey,
+  type UseMutationOptions,
+  type UseMutationResult,
+} from '@tanstack/react-query'
+
+/**
+ * Return a function that will execute the given mutation.
+ * Unlike `useMutation`, this function returns a function instead of the state of the mutation.
+ */
+export function useMutationCallback<
```

Comment:
very cool, i can guess why this is a lot faster...

... i wonder why tanstack query doesn't expose these directly...